gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactive/reactor/processor/forward/XForwardForProcessor.java

```diff
@@ -66,4 +69,54 @@ public Completable execute(final HttpExecutionContextInternal ctx) {
             }
         });
     }
+
+    static String generateOriginalUrl(HttpBaseRequest request) {
+        String forwardedHeaderValue = request.headers().get(HttpHeaderNames.FORWARDED);
+
+        if (!StringUtils.hasText(forwardedHeaderValue) && !StringUtils.hasText(request.headers().get(HttpHeaderNames.X_FORWARDED_FOR))) {
```

Comment:
Looks like it ignores the other possible headers in case there is no Forwarded and X-Forwarded-For header? I don't understand well this part.